### PR TITLE
feat(artifact): add artifact component

### PR DESCRIPTION
**Pull request by chuang8511**
Because

- we want users to be able to use catalog from VDP

This commit

- add artifact component

### CODE REVIEW

File: data/artifact/v0/client.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+	"google.golang.org/grpc/metadata"
+)
+
+const maxPayloadSize int = 1024 * 1024 * 32
+
+func initArtifactClient(serverURL string) (pbClient interface{}, connection interface{}, err error) {
+	var clientDialOpts grpc.DialOption
+
+	if strings.HasPrefix(serverURL, "https://") {
+		clientDialOpts = grpc.WithTransportCredentials(credentials.NewTLS(&tls.Config{InsecureSkipVerify: true}))
```
[the remark this answers is not in the document]
Comment (chuang8511) [reply data/artifact/v0/client.go:22]: else part is for CE version. So, it should be fine.

**Comment by chuang8511** [comment]
will resolve conflict when it is going to be reviewed.

**Comment by donch1989** [comment]
Hi @chuang8511 
I think we can merge before the query function are ready.

**Comment by chuang8511** [comment]
@donch1989 rebased

**Pull request merged**